### PR TITLE
Upgrade timeout version requirement

### PR DESCRIPTION
**Pull request by alexandreagular**
The pull request changes the version requirement for the setReceiveTimeOut, setSendTimeOut, getReceiveTimeOut and getSendTimeOut methods based on this:
http://api.zeromq.org/2-2:zmq-getsockopt

The minimal version required is now 2.2.0.

Also I added a unit test which fails at the moment. I haven't figured out yet where the problem is but i am working on it.

### CODE REVIEW

**Comment by alexandreagular** [comment]
I have solved the bug. It turns our that these options where set using long instead of ints, which was inconsistent with the ZeroMQ spec.

I will add some other tests for other options and make the appropriate fixes.

**Comment by jvirtanen** [comment]
Thanks, @alexandreagular! I will have a look into your changes later this week.

**Comment by alexandreagular** [comment]
Hi @jvirtanen, did you have the time to take a look at what I did ?

**Comment by jvirtanen** [comment]
Hi @alexandreagular, sorry that I didn't manage to go through these changes last week.

Could you rebase you changes on the latest `master`?

File: build.sbt [hunk 1]
```diff
@@ -4,12 +4,15 @@ name := "zeromq-scala-binding"
 
 version := "0.0.8-SNAPSHOT"
 
+//scalaBinaryVersion := "2.10.0-M7"
```
Comment (jvirtanen) [inline build.sbt]: Could you drop these changes from this pull request as they are unrelated?

File: build.sbt [hunk 2]
```diff
@@ -4,12 +4,15 @@ name := "zeromq-scala-binding"
...
+
 libraryDependencies ++= Seq(
   "net.java.dev.jna" %  "jna"             % "3.0.9",
   "com.github.jnr"   %  "jnr-constants"   % "0.8.2",
   //"ch.qos.logback"   %  "logback-classic" % "1.0.0", // uncomment to see logging output
   "org.slf4j"        %  "slf4j-api"       % "1.6.4",
-  "org.scalatest"    %% "scalatest"       % "1.6.1" % "test"
+  "org.scalatest"    %% "scalatest"       % "1.6.1" % "test" //To build with Scala 2.9.2
+  //"org.scalatest"    %% "scalatest"       % "1.9-2.10.0-M7-B1" % "test" //To build with Scala 2.10.0-M7
```
Comment (jvirtanen) [inline build.sbt]: See above.

**Comment by alexandreagular** [comment]
Hi,

I did the rebase. Also I checked again the unit tests and they pass.

I think it's good now, can you check ?

**Comment by alexandreagular** [comment]
Hi,

Could I have an update on this please ?

**Comment by alexandreagular** [comment]
Hi,

Could I have an update on this please ?

**Pull request closed without merging**